lib/lambda_ethereum_consensus/validator/validator.ex

```diff
@@ -35,29 +42,39 @@ defmodule LambdaEthereumConsensus.Validator do
   ### GenServer Callbacks
   ##########################
 
-  @impl true
-  def init({slot, head_root}) do
-    config = Application.get_env(:lambda_ethereum_consensus, __MODULE__, [])
-
-    validator =
-      case {Keyword.get(config, :pubkey), Keyword.get(config, :privkey)} do
-        {nil, nil} -> nil
-        {pubkey, privkey} -> %{index: nil, privkey: privkey, pubkey: pubkey}
-      end
+  @type state :: %{
+          slot: Types.slot(),
+          root: Types.root(),
+          duties: %{
+            attester: list(:not_computed | Types.epoch()),
+            proposer: :not_computed | list(Types.slot())
+          },
+          validator: any(),
+          payload_builder: {Types.slot(), Types.root(), BlockBuilder.payload_id()} | nil
+        }
 
+  @impl true
+  @spec init({Types.slot(), Types.root(), {Bls.pubkey(), Bls.privkey()}}) ::
+          {:ok, state, {:continue, any}}
+  def init({head_slot, head_root, {pubkey, privkey}}) do
     state = %{
-      slot: slot,
+      slot: head_slot,
```

Comment:
Probably something to think about in a separate issue/PR, but what happens if a validator process dies and is revived after the node has been running for a while? If I'm thinking this correctly, the initial head_slot and head_root will probably be highly outdated. Is that something that may cause issues here?

My intuition tells me that pubkey and privkey are effectively init args, while head_slot and head_root should probably be obtained from a single source of truth instead of being passed as init args.

Reply:
In general we should have a ticket to revisit restarts of genservers.

Reply:
https://github.com/lambdaclass/lambda_ethereum_consensus/issues/1097